v2/eventsource/stores/memorystore/store.go

```diff
@@ -2,12 +2,14 @@ package memorystore
 
 import (
 	"context"
+	"sync"
 
 	"github.com/SKF/go-eventsource/v2/eventsource"
 )
 
 type store struct {
 	Data map[string][]eventsource.Record
+	mutex  sync.RWMutex
```

Comment:
If you want to not have to care about mutexes you could use sync.Map instead of regular map + mutex. 
It is probably the thing to use except when you want very specific optimizations with read and write locks

Reply:
Sync.map is note very nice in go since we do not have generics :)